### PR TITLE
Implement Display for Ethertype and IpNextHeaderProtocol

### PR DESCRIPTION
**Pull request by fsck**
Also adds leading zeros to Display for MacAddr.

### CODE REVIEW

File: src/packet/ip.rs [hunk 1]
```diff
@@ -481,3 +482,167 @@ impl PrimitiveValues for IpNextHeaderProtocol {
...
+            &IpNextHeaderProtocols::MplsInIp => "MplsInIp", // 137
+            &IpNextHeaderProtocols::Manet => "Manet", // 138
+            &IpNextHeaderProtocols::Hip => "Hip", // 139
+            &IpNextHeaderProtocols::Shim6 => "Shim6", // 140
+            &IpNextHeaderProtocols::Wesp => "Wesp", // 141
+            &IpNextHeaderProtocols::Rohc => "Rohc", // 142
+            &IpNextHeaderProtocols::Test1 => "Test1", // 253
+            &IpNextHeaderProtocols::Test2 => "Test2", // 254*/
+            &IpNextHeaderProtocols::Reserved => "Reserved", // 255
+            _                          => "unknown"
```
Comment (mrmonday) [inline src/packet/ip.rs]: I'll trust you that this is exhaustive :wink:

**Comment by mrmonday** [comment]
Looks great, thanks!

@homu r+

File: src/packet/ip.rs (hunk 1 above)
Comment (fsck) [inline src/packet/ip.rs]: I guess [this](http://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml) is the official reference. I could not find any discrepancies. But if anybody has any complaints I am happy to provide an addendum. ;-)

Comment (mrmonday) [inline src/packet/ip.rs]: I originally generated all those constants from the list, so if you have them all (unless there are new entries), that should be all of them I think!

**Pull request merged**